public/app/features/visualization/data-hover/DataHoverView.tsx

```diff
@@ -41,9 +41,8 @@ export function getDisplayValuesAndLinks(
   sortOrder?: SortOrder,
   mode?: TooltipDisplayMode | null
 ) {
-  const fields = data.fields.map((f, idx) => {
-    return { ...f, hovered: idx === columnIndex };
-  });
```

Comment:
this field copy is the root cause of why `===` was failing inside `getFieldDisplayName()`, and resulting in the `1` suffix, which is the original issue https://github.com/grafana/grafana/pull/75511 was attempting to fix.